src/Microsoft.OpenApi.OData.Reader/Common/EdmModelHelper.cs

```diff
@@ -83,7 +84,91 @@ internal static OpenApiSchema GetDerivedTypesReferenceSchema(IEdmStructuredType
             // if the individual has no navigability setting, use the global navigability setting
             // Default navigability for all navigation properties of the annotation target.
             // Individual navigation properties can override this value via `RestrictedProperties/Navigability`.
-            return restrictionProperty?.Navigability != null || restrictionType == null || restrictionType.IsNavigable;            
+            return restrictionProperty?.Navigability != null || restrictionType == null || restrictionType.IsNavigable;
+        }
+
+        /// <summary>
+        /// Generates the operation id for a navigation property path.
+        /// </summary>
+        /// <param name="path">The target <see cref="ODataPath"/>.</param>
+        /// <param name="navigationSource">The <see cref="IEdmNavigationSource"/> of the target path.</param>
+        /// <param name="prefix">Identifier indicating whether it is a collection-valued non-indexed navigation property.</param>
+        /// <returns>The operation id name.</returns>
+        internal static string GenerateNavigationPropertyPathOperationId(ODataPath path, IEdmNavigationSource navigationSource, string prefix = null)
```

Comment:
Can we standardize the naming convention of operationIds to what we have in DevX API - https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/dev/OpenAPIService/PowershellFormatter.cs?
For example:
- Remove hash suffix values from operationIds of function paths.
- Add '_' to separate verb (action) in an operationId. This typically the last `.` or second `.` for OData cast paths.
- PUT operations should have Set as the verb in an operationId `{xxx}_Set{yyy}`.

In PowerShell, we use operationIds to form command names.

Reply:
See my comments at https://github.com/microsoft/OpenAPI.NET.OData/issues/324#issuecomment-1419518641 on operationIds and tags for count and complex property paths.

Reply:
> Can we standardize the naming convention of operationIds to what we have in DevX API - https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/dev/OpenAPIService/PowershellFormatter.cs? For example:
> 
> * Remove hash suffix values from operationIds of function paths.
> * Add '_' to separate verb (action) in an operationId. This typically the last `.` or second `.` for OData cast paths.
> * PUT operations should have Set as the verb in an operationId `{xxx}_Set{yyy}`.
> 
> In PowerShell, we use operationIds to form command names.

There are other clients who rely on the operation ids, not sure it will be a good idea to handle that in this library. See this [issue](https://github.com/microsoftgraph/msgraph-metadata/issues/289).

Reply:
They are also using AutoREST in the referenced issue - https://github.com/microsoftgraph/msgraph-metadata/issues/289. Standardizing the operationIds here and ensuring uniqueness will have the net effect of fixing their issue.

To unblock this change, I've created an issue at https://github.com/microsoft/OpenAPI.NET.OData/issues/361 for us to discuss this further.